### PR TITLE
[json]: add docs to `hardfork specification`

### PR DESCRIPTION
**Pull request by niklasad1**
Fix https://github.com/paritytech/parity-ethereum/pull/11039#discussion_r331434940

### CODE REVIEW

File: json/src/spec/spec.rs [hunk 1]
```diff
@@ -24,17 +24,29 @@ use serde_json::Error;
...
 	Frontier,
+	/// Homestead (#1,150,000, 2016-03-14)
 	Homestead,
+	/// EIP 150 Tangerine Whistle: Gas cost changes for IO-heavy operation (#2,463,000, 2016-10-18)
+	EIP150,
+	/// EIP 158/EIP 161 Spurious Dragon: State trie clearing (#2,675,000, 2016-11-22)
+	EIP158,
+	/// Byzantium Metropolis phase 1 (#4,370,000, 2017-10-16)
 	Byzantium,
+	/// Constantinople Metropolis phase (#7,280,000, 2019-02-28)
```
Comment (ordian) [inline json/src/spec/spec.rs]: phase 2?

File: json/src/spec/spec.rs [hunk 2]
```diff
@@ -24,17 +24,29 @@ use serde_json::Error;
 /// Fork spec definition
 #[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Deserialize)]
 pub enum ForkSpec {
-	EIP150,
-	EIP158,
```
Comment (ordian) [inline json/src/spec/spec.rs]: `ForkSpec` derives `Ord`, I don't know if changing the order here breaks something or not?

File: json/src/spec/spec.rs (hunk 1 above)
Comment (niklasad1) [reply json/src/spec/spec.rs]: yes

File: json/src/spec/spec.rs (hunk 2 above)
Comment (niklasad1) [reply json/src/spec/spec.rs]: reverted

**Pull request merged**